bin/nvidia-container-toolkit

```diff
@@ -26,6 +26,33 @@ device_wait() {
 
 }
 
+cdi_generate () {
+    # Generate the CDI config #
+    "${SNAP}/usr/bin/nvidia-ctk" cdi generate --nvidia-ctk-path "${SNAP}/usr/bin/nvidia-ctk" --output="${SNAP_DATA}/etc/cdi/nvidia.yaml"
+}
+
+runtime_configure () {
+    # Generate the dockerd runtime config #
+    "${SNAP}/usr/bin/nvidia-ctk" runtime configure --runtime=docker --runtime-path "${SNAP}/usr/bin/nvidia-ctk" --config "${SNAP_DATA}/config/daemon.json"
+}
+
+setup_fail () {
+    echo "WARNING: Conainter Toolkit setup seemed to fail with an error"
+
+    # Remove nvidia runtime config, if it exists #
+    jq -r 'del(.runtimes.nvidia)' "${SNAP_DATA}/config/daemon.json" > "${SNAP_DATA}/config/daemon.json.new"
+
+    # If it was removed [ there was a change ], copy in the new config, remove CDI config,  and set service restart flag #
+    if ! cmp "${SNAP_DATA}/config/daemon.json"{,.new} >/dev/null ; then
+        mv "${SNAP_DATA}/config/daemon.json"{.new,}
+        rm -f "${SNAP_DATA}/etc/cdi/nvidia.yaml"
+    fi
+}
+
+setup_info () {
+    echo "Conainter Toolkit setup complete"
+}
+
```

Comment:
I wonder if nvidia support should be disabled with snap set upon failing to create it. As likely kernel & toolkit libraries refresh would be required to resolve it.

Reply:
>As likely kernel & toolkit libraries refresh would be required to resolve it.

@xnox 

That's true.  It would then also require specific action to re-enable however.

It's hard to try and anticipate all different end user scenarios, but I think on balance not blocking install is best.  In the scenarios where non-nvidia containers are also required this doesn't block that.

The only other option I can think of is to make this behaviour configurable.  Something like:
```
nvidia-support.strict: true|false
```
- `true`
  - install fails if nvidia support fails to be configured
- `false`
  - install continues if nvidia support fails to be configured

What do do think ?